docs/quickstart.rst

```diff
@@ -28,59 +31,128 @@ Lets get started from the feature engineering in `Spark`_.
-        .config('spark.jars.packages', 'org.apache.hadoop:hadoop-aws:2.7.4')
-        .master("local[*]")
         .getOrCreate()
     )
 
     df = spark.createDataFrame(
         [
             {
                 "id": 1,
-                "mat": DenseMatrix(2, 2, range(4)),
                 "image": Image("s3://foo/bar/1.png"),
                 "annotations": [
                     Row(
                         text="cat",
+                        label=2,
                         mask=np.random(size=(256, 256)),
-                        bbox=Box2d(x=1.0, y=2.0, width=3.0, height=4.0)
+                        bbox=Box2d(xmin=1.0, ymin=2.0, xmax=3.0, ymax=4.0)
                     )
                 ]
             },
         ]
     )
 
-    df.write.format("rikai").save("dataset/out")
+    df.write.format("rikai").save("my_dataset")
 
-We can then inspect the dataset in a `Jupyter Notebook`_.
+The magic here is that Rikai maintains commonly used :doc:`Semantic Types <types>`
+and takes care of ``SerDe`` (short for **Ser**-ialization and **De**-serialization)
+and visualization in notebooks.
+
+Additionally, Rikai community maintains a set of pre-baked connectors,
+such as `COCO <https://cocodataset.org/#home>`_ and `ROS Bag <http://wiki.ros.org/Bags>`_
+
+When it is ready, we can submit the script via ``spark-submit``
+
+.. code-block:: bash
 
+    spark-submit \
+      --master yarn \
+      --packages ai.eto:rikai_0.12:0.0.16 \
+      script.py
 
 
+Step 2. Inspect Dataset
+------------------------
+
+We can then inspect the dataset in a `Jupyter Notebook`_.
+
 .. code-block:: python
 
-    df = spark.read.format("rikai").load("dataset/out")
+    df = spark.read.format("rikai").load("my_dataset")
     df.printSchema()
     df.show(5)
 
 
+Step 3. Train the model
+-----------------------
 
-Use the dataset in `pytorch`
+Use this dataset in `Pytorch`_
 
 .. code-block:: python
 
-    from rikai.torch import DataLoader
+    import torch
+    import torchvision
+    from rikai.torch.vision import Dataset
+    from torch.utils.data import DataLoader
+
+    device = torch.device("cuda") if \
+        torch.cuda.is_available() else torch.device("cpu")
+
+    dataset = Dataset(
+        "my_dataset",
+        image_column="image",
+        target_column='annotations',
+        transform=torchvision.transforms.ToTensor(),
+    )
 
     data_loader = DataLoader(
-        "dataset/out",
-        shuffle=True,
-        batch=8,
+        dataset,
+        batch_size=8,
+        num_workers=4,
     )
-    for examples in data_loader:
-        print(example)
+
+    model.train()
+    for epoch in range(10):
+        for imgs, annotations in data_loader:
+            loss_dict = model(imgs, annotations)
+            losses = sum(loss_dict.values())
+            optimizer.zero_grad()
+            losses.backward()
+            optimizer.step()
+
+Rikai offers `MLflow`_ integration. When a model registered with `MLflow`, it will be available
```

Comment:
Should this be a separate section about "model registry" ?

Reply:
So what I was thinking is , in `Quickstart`, it describe a "user workflow", so "training" from user's aspect, is part of a development workflow.

And we will have separate sections to describe functions / components. i.e., all UDFs,  how to configure MLflow / Model registry, how to use semantic types' visualization capability in notebook and etc.